mrp_client.py

```diff
@@ -19,56 +18,64 @@ def __init__(self, parser, args):
             self.log.fatal(err)
             exit(1)
 
-    def getip(self, machine_name, interface_name='eth1'):
-        try:
-            ipgetter = IPGetter(self.urlhandler, machine_name, interface_name)
-            return ipgetter.get_ip()
-        except Exception as err:
-            self.log.fatal(err)
-            exit(1)
+    def parse(self):
+        if self.args.subcommand == 'image':
+            self.image_upload(self.args.action, self.args.image_type, self.args.description,
+                              self.args.arch, self.args.image_path, self.args.public,
+                              self.args.knowngood)
+        else:
+            self.parser.print_help()
 
-    def getmac(self, machine_name, interface_name='eth1'):
-        try:
-            ipgetter = IPGetter(self.urlhandler, machine_name, interface_name)
-            return ipgetter.get_mac()
-        except Exception as err:
-            self.log.fatal(err)
-            exit(1)
 
-    def getnetmask(self, machine_name, interface_name='eth1'):
+    def image_upload(self, command, image_type, desc, arch, path, public, knowngood):
+        image_uploader = ImageControl(self.urlhandler)
         try:
-            ipgetter = IPGetter(self.urlhandler, machine_name, interface_name)
-            return ipgetter.get_netmask()
+            if command == 'upload':
+                rc = image_uploader.upload_image(image_type, desc, arch, path, public,
+                                                 knowngood)
+                self.log.debug(rc)
+            elif command == 'check':
+                if image_uploader.check_for_existence(image_type, desc, arch):
+                    print('True')
+                else:
+                    print('False')
+                    exit(2)
```

Comment:
Ah, yes, that's good. We return true/false and non-zero on error, so that both unix and scripts can interface with it.

Reply:
Yes ! A non-zero (2) different from a fatal exception (1) !